core/src/main/java/org/opensearch/sql/executor/execution/AbstractPlan.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sql.executor.execution;
+
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.opensearch.sql.common.response.ResponseListener;
+import org.opensearch.sql.executor.ExecutionEngine;
+import org.opensearch.sql.executor.QueryId;
+
+/**
+ * AbstractPlan represent the execution entity of the Statement.
+ */
+@RequiredArgsConstructor
+public abstract class AbstractPlan {
+
+  /**
+   * Uniq query id.
+   */
+  @Getter
+  private final QueryId queryId;
+
+  /**
+   * Start query execution.
+   */
+  public abstract void execute();
+
+  /**
+   * Explain query execution.
+   *
+   * @param listener query explain response listener.
+   */
+  public abstract void explain(ResponseListener<ExecutionEngine.ExplainResponse> listener);
```

Comment:
Just comment here in case we forget: I think we need to merge Explain statement to single query plan structure (probably done in QueryPlanFactory?). In this way, all DQL, DML, DDL, Explain/Analyze will be same QueryPlan with plan tree and execute() method. No need to differentiate special plan like explain in AbstractPlan, QueryService, Response listener and RestAction.

Reply:
Agree, add an issue to track. https://github.com/opensearch-project/sql/issues/985